File: app/components/folio/console/ui/collapsible_component.js
window.Folio.Stimulus.register('f-c-ui-collapsible', class extends window.Stimulus.Controller {
  static values = {
    collapsed: { type: Boolean, default: true }
  }

  onToggleClick (e) {
    e.preventDefault()
    this.collapsedValue = !this.collapsedValue

    if (!this.collapsedValue) {
      const input = this.element.querySelector('.f-c-ui-collapsible-focus-input')
      if (input) input.focus()
    }
  }
})

File: app/components/folio/console/ui/collapsible_component.rb
# frozen_string_literal: true

class Folio::Console::Ui::CollapsibleComponent < Folio::Console::ApplicationComponent
  FOCUS_INPUT_CLASS_NAME = "f-c-ui-collapsible-focus-input"

  def initialize(title:, collapsed: true)
    @title = title
    @collapsed = collapsed
  end

  private
    def data
      stimulus_controller("f-c-ui-collapsible",
                          values: {
                            collapsed: @collapsed,
                          })
    end
end
